src/Integration/NewConnectedMode/ConfigurationProvider.cs

```diff
@@ -70,8 +70,10 @@ public bool Persist(BoundSonarQubeProject project, SonarLintMode bindingMode)
 
             var writeSettings = GetWriteSettings(bindingMode);
 
-            return writeSettings.HasValue && 
+            var success = writeSettings.HasValue && 
                    solutionBindingSerializer.Write(writeSettings?.ConfigPath, project, writeSettings?.OnSuccessfulFileWrite);
+
+            return success ? TryGetBindingConfiguration(writeSettings?.ConfigPath, project, bindingMode) : null;
```

Comment:
solutionBindingSerializer.Write only queues the writing operation, so I couldn't call GetConfiguration. I had to instead use the created `project` and initialize a new BindingConfiguration object.

Reply:
As discussed offline, I think that's ok. We need to create a new instance somewhere.